funzione per la lettura della geometria

davidair8 opened the pull request:
The current release contains:
- a standard text file 'propgeometry.txt' that the user has to compile with the informations about the propeller geometry;
- a Matlab class 'geometryreader' that allocates the informations of the text file in local variables. This class also contains two 
 functions:
   -"xRotorAnalisys" that communicates with 'xrotor.exe' to analize the propeller and saves the results in local variables ;
   -"plotResults" that also plot the results.
- an example Matlab script 'main' that shows how the user can request to the 'geometryreader' class the desired operations.

Commit:
La release contiene 4 file:
-un file testo standard 'propgeometry.txt' che l'utente deve compilare con le informazioni sulla geometria della pala;
-una class Matlab 'geometryreader' che alloca in variabili locali le informazioni del file testo e, se richiesto dall'utente, si interfaccia con xrotor per analizzare l'elica;
-un file Matlab 'main' dal quale l'utente può richiedere alla class le operazioni desiderate;
-l'eseguibile di xrotor.

EttoreSaetta commented on geometryreader.m:
```diff
@@ -0,0 +1,281 @@
+%----------------------------------------------------------------------
```
Qui va inserito l'header standard che avete messo nel main

EttoreSaetta commented on main.m:
```diff
@@ -0,0 +1,69 @@
+%% \file funzione per la lettura della geometria
```
Il main serve per chimare la funzione che avete scritto per verificarne la funzionalità.
Quindi non serve che il main sia presente nella libreria.
Al massimo potete aggiungere una cartella "TestCases" ed una sotto cartella "Lettura geometria" (ad esempio) che mostri come utilizzare la funzione tramite questo main

EttoreSaetta commented on propgeometry.txt:
```diff
@@ -0,0 +1,36 @@
+----------------------------------------------------------------------
```
Anche in questo caso sarebbe utile inserire questo file di testo nella cartella "TestCases"

EttoreSaetta reviewed (commented):
Alcune modifiche preliminari da fare.
Il file con l'eseguibile di xrotor non va inserito. Deve essere l'utente che utilizza la libreria ad avere già xrotor installato.
Ricordate che qualunque file aggiungete deve contenere l'header con la descrizione e con la licenza.

Commit:
xrotor executable has been removed, moreover two more folders have been created:
-Test Cases that contains the standard .txt file with the geometry informations
-the Test Cases subfolder: lettura geometria that contains the main.m file with the istructions for the users.

EttoreSaetta commented:
Sposterei anche il file .txt nella sottocartella di Test Cases.
Inoltre, non lasciate spazi nei nomi delle cartelle, questo potrebbe creare problemi se utilizzato tramite riga di comando linux.
Quindi cambiate i nomi in TestCases e GeometryReader.

Commit:
Riorganizzazione e rinomina delle cartelle.

EttoreSaetta commented on TestCases/GeometryReader/main.m:
```diff
@@ -0,0 +1,69 @@
+lastadv='0.363';   %last value of the advance ratio vector of the analysis
+advstep='0.003';   %step of the advance ratio
+analize=1;
+plots=0;
```
It is better to put also these data in the input file, maybe crating a new section in the input file named analysis parameters.
In this way they are not `global` anymore in your `geometryreader.m`.

EttoreSaetta reviewed (commented):
Just a few corrections, in general it looks good to me.
You have to indent the code.
In order to improve readability, it is better to put the analysis zone and the plot zone in two different functions. These new functions should be in the same file of the geometryreader.m, because you defined a class. So that there will be 3 functions in your class:
1. Input read
2. xRotor analysis
3. Plot results

In this way it will also be more easy for the others to call only the function that reads the input text file.

Commit:
New release 29/12/2020. 

Two new functions have been added to the class "geometryreader":
-"xRotorAnalysis" that analyzes the geometry via xrotor;
-"plotResults" that generates the plots of the analysis results.

The variables "lastadv", "firstadv", "advstep" are no more global: user has to set them in the input file.

Commit:
Header has been added.

EttoreSaetta reviewed (commented):
@davidair8
You also should provide a documentation of your function. You can do it after or before the merge.
If you want to provide the documentation after the merge, you will have to open a new PR just for the documentation.
Therefore I suggest to provide the documentation first.
Let me know what do you want to do.

Commit:
Lock number and equivalent parasitic area have been added to the propgeomrtry.txt file. Documentation has been produced.

EttoreSaetta commented:
@davidair8 The name of the documentation folder should be `doc`.
The name of the documentation file should be clear: use something like `geometry_reader.pdf`.

Commit:
The folders have been renamed.

EttoreSaetta reviewed (approved):
LGTM
Merge it whenever you are ready.
